tests/test_cache.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+from pathlib import Path
+import h5py
+import pandas as pd
+import pytest
+import numpy as np
+from loguru import logger
+import celeri
+from celeri.operators import _OperatorBuilder, _store_elastic_operators, _hash_elastic_operator_input
+
+test_logger = logger.bind(name="test_output_files")
+
+@pytest.mark.parametrize(
+    "config_file",
+    [
+        "tests/configs/wna_outputs.json",
+    ],
+)
+def test_smart_segment_recompute(config_file):
+    """Test that selective recompute works correctly when segments change.
+    
+    This test verifies the selective recomputation feature of elastic operators:
+    1. First computes operators from scratch and caches them
+    2. Modifies a segment file and triggers selective recompute (only changed segments)
+    3. Renames the cache file and recomputes from scratch
+    4. Compares the selective recompute results with full recompute to ensure they match
+    
+    Args:
+        config_file: Path to the test configuration file
+    """
+    logger.disable("celeri")
+    cache_dir = None
+    try:
+        test_logger.info("Computing operator with original segment file")
+        model = celeri.build_model(config_file)
```

Comment:
I think it would be better to create a config object first, change the cache dir to a tempdir from `tempfile`, and create the model with that modified config. That way we control the directory directly, and can simplify this code a bit?